UefiCpuPkg/Library/MmUnblockMemoryLib/MmUnblockMemoryLib.c

```diff
@@ -45,12 +127,20 @@ MmUnblockMemoryRequest (
   EFI_STATUS                    Status;
   MM_UNBLOCK_REGION             *MmUnblockMemoryHob;
   EFI_PEI_MM_COMMUNICATION_PPI  *MmCommunicationPpi;
+  EFI_PEI_HOB_POINTERS          Hob;
 
   if (!IS_ALIGNED (UnblockAddress, SIZE_4KB)) {
     DEBUG ((DEBUG_ERROR, "Error: UnblockAddress is not 4KB aligned: %p\n", UnblockAddress));
     return EFI_INVALID_PARAMETER;
   }
 
+  Hob.Raw = GetFirstHob (EFI_HOB_TYPE_MEMORY_ALLOCATION);
+  ASSERT (Hob.Raw != NULL);

```

Comment:
Move the GetFirstHob() call inside the sub-function.

Reply:
GetHobList () is passed into sub-function here